detection_rules/semver.py

```diff
@@ -4,14 +4,21 @@
 
 """Helper functionality for comparing semantic versions."""
 import re
+from typing import Iterable, Union
 
 
 class Version(tuple):
 
-    def __new__(cls, version):
+    def __new__(cls, version: Union[Iterable, str], pad: int = None) -> 'Version':
```

Comment:
Do you still need this?
Also if you want an optional integer w/ a default of None, you're looking for `typing.Optional[int] = None`